Plugins/Format Source Code/plugin.swift

```diff
@@ -0,0 +1,93 @@
+import PackagePlugin
+import Foundation
+
+@main
+struct FormatSourceCode: CommandPlugin {
+  func performCommand(
+    context: PluginContext,
+    arguments: [String]
+  ) async throws {
+    let swiftFormatTool = try context.tool(named: "swift-format")
+    
+    // Extract the arguments that specify what targets to format.
+    var argExtractor = ArgumentExtractor(arguments)
+    let targetNames = argExtractor.extractOption(named: "target")
+    let targetsToFormat = try context.package.targets(named: targetNames)
+    
+    let configuration = argExtractor.extractOption(named: "configuration").first
+    
+    for target in targetsToFormat {
+      // Skip any type of target that doesn't have source files.
+      guard let target = target as? SourceModuleTarget else {
+        Diagnostics.warning("Skipping target “\(target.name)” because it doesn't contain Swift sources")
+        continue
+      }
+      
+      let swiftFormatExec = URL(fileURLWithPath: swiftFormatTool.path.string)
+      var swiftFormatArgs = [
+        "format",
+        target.directory.string,
+        "-r",
+        "-p",
+        "-i",
+      ]
+      
+      if let configuration {
+        swiftFormatArgs.append("--configuration")
+        swiftFormatArgs.append(configuration)
+      }
+      
+      let process = try Process.run(swiftFormatExec, arguments: swiftFormatArgs)
+      process.waitUntilExit()
+      
+      // Check whether the subprocess invocation was successful.
+      if process.terminationReason == .exit && process.terminationStatus == 0 {
+        print("Formatted the source code in \(target.directory).")
+      }
+      else {
+        let problem = "\(process.terminationReason):\(process.terminationStatus)"
+        Diagnostics.error("swift-format invocation failed: \(problem)")
+      }
+    }
+  }
+}
+
+#if canImport(XcodeProjectPlugin)
+import XcodeProjectPlugin
+
+extension FormatSourceCode: XcodeCommandPlugin {
+  func performCommand(context: XcodeProjectPlugin.XcodePluginContext, arguments: [String]) throws {
+    let swiftFormatTool = try context.tool(named: "swift-format")
+    
+    var argExtractor = ArgumentExtractor(arguments)
+    let configuration = argExtractor.extractOption(named: "configuration").first
+    
+    let swiftFormatExec = URL(fileURLWithPath: swiftFormatTool.path.string)
+    var swiftFormatArgs = [
+      "format",
+      ".",
```

Comment:
From the `XcodePluginContext` you can reach through `context.xcodeProject.directory`, which might be safer than assuming the working directory of the plugin will always be the project directory.

Reply:
I fix it. https://github.com/zunda-pixel/swift-format/commit/2fee7b9c1ad81955a054c02e6ca9884246b5d2b5